### PR TITLE
MBS-9825 ability to test runner in integration

### PR DESCRIPTION
**Pull request by dsvoronin**

### CODE REVIEW

File: subprojects/gradle/ci-logger/src/main/kotlin/com/avito/utils/logging/Gradle.kt [hunk 1]
```diff
@@ -67,7 +69,13 @@ private fun defaultCILogger(
         debugHandler = CILoggingCombinedHandler(
             handlers = listOf(
                 destinationFileHandler
-            )
+            ).let {
+                if (isInvokedFromIde) {
+                    it.plus(onlyMessageStdoutHandler)
```
Comment (dsvoronin) [inline subprojects/gradle/ci-logger/src/main/kotlin/com/avito/utils/logging/Gradle.kt:74]: debug level also goes to stdout, but only from IDE invocations

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt [hunk 2]
```diff
@@ -82,14 +82,19 @@ abstract class InstrumentationConfiguration(val name: String) {
...
             val hasKubernetes = targets.any { it.reservation.device is Device.CloudEmulator }
+            val hasMockEmulator = targets.any { it.reservation.device is Device.MockEmulator }
             if (hasLocal && hasKubernetes) {
-                throw IllegalStateException("Targeting to local and kubernetes emulators at the same configuration $name is not supported yet")
+                throw IllegalStateException("Targeting local and kubernetes emulators at the same configuration $name is not supported")
+            } else if (hasMockEmulator && (hasLocal || hasKubernetes)) {
+                throw IllegalStateException("Targeting mock emulators and has local or kubernetes emulators at the same configuration $name is not supported")
             } else {
                 isTargetLocalEmulators = hasLocal
+                isMockEmulator = hasMockEmulator
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt]: two properties to enum

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/LocalDevicesProvider.kt [hunk 3]
```diff
@@ -45,8 +44,11 @@ class LocalDevicesProvider(
                 launch {
                     do {
                         val acquiredCoordinates = mutableSetOf<DeviceCoordinate>()
-                        val findDevices = findDevices(reservation, acquiredCoordinates)
-                        findDevices.forEach { device ->
+                        val adbDevices = findDevices(reservation, acquiredCoordinates)
+
+                        logger.info("Found local devices: $adbDevices")
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/LocalDevicesProvider.kt]: debug

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/MockDevicesProvider.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.concurrent.TimeUnit
+
+class MockDevicesProvider(private val logger: CILogger) : DevicesProvider {
+
+    private val devices = Channel<Device>(Channel.UNLIMITED)
+
+    @ExperimentalCoroutinesApi
+    override fun provideFor(reservations: Collection<Reservation.Data>, scope: CoroutineScope): ReceiveChannel<Device> {
+        val devicesRequired = reservations.fold(0, { acc, reservation -> acc + reservation.count })
+        scope.launch(Dispatchers.IO) {
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/MockDevicesProvider.kt:29]: constructor

File: subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/mock/MockDevice.kt [hunk 5]
```diff
@@ -1,25 +1,34 @@
...
 import java.nio.file.Path
 import java.util.ArrayDeque
+import java.util.Date
 import java.util.Queue
 
 open class MockDevice(
-    override val coordinate: DeviceCoordinate,
-    private val logger: Logger,
+    override val coordinate: DeviceCoordinate = randomDeviceCoordinate(),
+    private val logger: Logger = object : Logger {
```
Comment (sergioboi) [inline subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/mock/MockDevice.kt]: emptyLogger?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasNotReportedTestsDeterminer.kt [hunk 6]
```diff
@@ -27,17 +28,19 @@ interface HasNotReportedTestsDeterminer {
...
             allTests: List<TestStaticData>
-        ): Result =
-
-            runResult.fold(
+        ): Result {
+            return runResult.fold(
                 { reportedTest ->
                     val allReportedTests = reportedTest.map { TestStaticDataPackage.fromSimpleRunTest(it) }
 
+                    log("allTests: $allTests")
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/HasNotReportedTestsDeterminer.kt]: delete?

File: subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/mock/MockDevice.kt (hunk 5 above)
Comment (dsvoronin) [reply subprojects/gradle/runner/shared-test/src/main/kotlin/com/avito/runner/test/mock/MockDevice.kt]: Found `NoOpLogger : Logger` in `:logger-test-fixtures`
not available here, 
i removed default value

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/MockDevicesProvider.kt (hunk 4 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/devices/provider/MockDevicesProvider.kt:29]: won't do in this PR scope
also needed in LocalDevicesProvider; better to change simultaneously

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt]: done

**Review by sergioboi — APPROVED** [review]

**Pull request merged**